### DISCUSSION TITLE
Bing results

### DISCUSSION THREAD

**Discussion by r2ixuz** [searxng/searxng#1209] [Q&A]
Why is every Bing result like a hashed Bing URL (https://www.bing.com › ck › a) ? I did not find any explanation in the docs.

**Comment by dalf** [comment]
I confirm the issue on my private instance and my laptop (4G mobile connection).

**Reply by dalf** [reply]
More information: the links to the same URL change on each query.

For example two links to searx.space: 

https://www.bing.com/ck/a?!&&p=20bed4d27e9cbbeb937130f699902952beca79af693483781afed1275a56a185JmltdHM9MTY1MjYzOTQyOCZpZ3VpZD0zMTBhMTY3OS04ZWM0LTQxMzAtOWVhZC0wYjlkNWM3YThlZGMmaW5zaWQ9NTE3MA&ptn=3&fclid=183e4fc6-d47d-11ec-8a7b-765cfc7c8507&u=a1aHR0cHM6Ly9zZWFyeC5zcGFjZS8&ntb=1

https://www.bing.com/ck/a?!&&p=5c80950749a24fe6a0ed0e5ba93b1488108485bd2b5a810bb368b5830caf66c9JmltdHM9MTY1MjYzOTQ0MSZpZ3VpZD00YTY1ZGQ5Yy05YWI4LTRiNGItOGNhNS0yMmE1YzhkNmRiMTUmaW5zaWQ9NTE3MA&ptn=3&fclid=1ff08e13-d47d-11ec-a595-16b2a5272406&u=a1aHR0cHM6Ly9zZWFyeC5zcGFjZS8&ntb=1


Using JS, the first link redirects to 
https://www.bing.com/ck/a?!&&p=20bed4d27e9cbbeb937130f699902952beca79af693483781afed1275a56a185JmltdHM9MTY1MjYzOTQyOCZpZ3VpZD0zMTBhMTY3OS04ZWM0LTQxMzAtOWVhZC0wYjlkNWM3YThlZGMmaW5zaWQ9NTE3MA&ptn=3&fclid=183e4fc6-d47d-11ec-8a7b-765cfc7c8507&u=a1aHR0cHM6Ly9zZWFyeC5zcGFjZS8&ntb=F

Which redirects to https://searx.space/ using the `location` header.

**Reply by mrpaulblack** [reply]
Also: https://github.com/searxng/searxng/issues/941

**Comment by return42** [comment]
Seems bing does no longer deliver direct links / disable JS, open https://www.bing.com/search?q=SearXNG and try to follow a link .. IDK how we can fix this in our bing engine.

**Reply by return42** [reply]
I tried to read URL from the (green) `<cite>` tag ..

```diff
     for result in eval_xpath(dom, '//li[@class="b_algo"]'):
 
+        url = extract_text(eval_xpath(result, './/cite'), allow_none=True)
+        if not url:
+            continue
         link = eval_xpath(result, './/h2/a')[0]
-        url = link.attrib.get('href')
```

But long URLs are ending with `...` --> e.g. `https://www.techtarget.com/searchapparchitecture/definition/foo-in...`

![grafik](https://user-images.githubusercontent.com/554536/168560451-a6a84b96-f909-4b99-9938-affe82dffbe9.png)

**Comment by jt196** [comment]
Can also confirm this is happening, in addition, the results are duplicated. So you'll have the direct link, then a Bing link often close by.

**Comment by dalf** [comment]
See https://github.com/searxng/searxng/pull/1219
